### PR TITLE
Update to `wgpu 0.20`, fixing crashes with some linux setups

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* Part of #6170
* Based on https://github.com/emilk/egui/pull/4433
* Makes it possible to handle https://github.com/rerun-io/rerun/issues/5968
    * it no longer is a crash but an _okish_ error message (more to do still)
 * Fixes https://github.com/rerun-io/rerun/issues/5283
    * [x] @jleibs can you confirm?

Test on:
* Mac
    * [x] native
    * [x] Firefox WebGL
    * [x] Chrome WebGPU
    * [x] Chrome WebGL 
* Windows
    * [x] native
    * [x] Firefox WebGL
    * [x] Chrome WebGPU
    * [x] Chrome WebGL 
* Linux
    * [x] native
    * [x] Firefox WebGL
    * [x] Chrome WebGPU
         * needs both chrome://flags/#enable-unsafe-webgpu & chrome://flags/#enable-vulkan flags set (via https://developer.chrome.com/docs/web-platform/webgpu/troubleshooting-tips)
    * [x] Chrome WebGL 

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6171?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6171?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6171)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/re_renderer/src/error_handling/error_tracker.rs [hunk 1]
```diff
@@ -106,7 +106,11 @@ impl ErrorTracker {
             wgpu::Error::OutOfMemory { source: _ } => {
                 re_log::error!("A wgpu operation caused out-of-memory: {error}");
             }
-            wgpu::Error::Validation {
+            wgpu::Error::Internal {
+                source: _source,
+                description,
+            }
+            | wgpu::Error::Validation {
```
Comment (emilk) [inline crates/re_renderer/src/error_handling/error_tracker.rs:113]: Don't we care about the difference between these?

Comment (Wumpf) [reply crates/re_renderer/src/error_handling/error_tracker.rs:113]: Hmmmm I thought no, but I think now you're actually right. `Validation` means we messed up with the inputs we gave to wgpu/webgpu whereas `Internal` means that either the wgpu/webgpu messed up and/or a driver isn't behaving as expected

**Comment by Wumpf** [comment]
Given the amount of failures and fixes we want to wait for wgpu 0.20.1

**Comment by Wumpf** [comment]
essentially waiting for https://github.com/gfx-rs/wgpu/pull/5681 now

**Comment by emilk** [comment]
I'm gonna split this branch/PR in two: one to track wgpu 0.20, one to update to egui master.
After this is merged:
* https://github.com/emilk/egui/pull/4559

**Comment by emilk** [comment]
I'll rebase this

**Review by Wumpf — COMMENTED** [review]
net 3 (small) dependencies less across the project (via cargo.lock: base64, objc_exception, safemem), neat

Did a few more changes to error_tracker.rs since I noticed that it didn't handle web correctly now that `webgpu` cfg is not a thing anymore

**Pull request merged**